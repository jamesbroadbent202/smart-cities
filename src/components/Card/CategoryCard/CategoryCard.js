import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Card from '../Card';
import style from './CategoryCard.scss';
import {
  CARD_SIZES,
  INDICATOR_CARD_TYPES,
} from '../../../constants';

const CategoryCard = (props) => {
  const className = classnames(
    style.wrapper,
    props.className,
  );

  return (
    <Card
      {...props}
      size={CARD_SIZES.LARGE}
      type={INDICATOR_CARD_TYPES.CATEGORY}
      className={className}
    >
      <div className={style.iconWrapper} />

      <div className={style.text}>{props.header}</div>
    </Card>
  );
};

CategoryCard.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string.isRequired,
  header: PropTypes.string,
  number: PropTypes.string,
  prefix: PropTypes.string,
  suffix: PropTypes.string,
};

export default CategoryCard;
