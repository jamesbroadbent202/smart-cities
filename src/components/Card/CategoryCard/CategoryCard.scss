@import '~style/variables';

.wrapper {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  padding: 20px;
}

.iconWrapper {
  // TODO (davidg): implement real icons
  border: 1px solid $color-white;
  width: 60px;
  height: 60px;
}

.text {
  @include type-indicator-card-title--large;

  margin-top: auto; // push to bottom
}
