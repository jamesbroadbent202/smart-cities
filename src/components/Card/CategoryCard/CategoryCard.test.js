import React from 'react';
import renderer from 'react-test-renderer';
import CategoryCard from './CategoryCard';

it('should match Snapshot', () => {
  const component =  renderer.create(
    <CategoryCard
      color="#eee"
    />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
