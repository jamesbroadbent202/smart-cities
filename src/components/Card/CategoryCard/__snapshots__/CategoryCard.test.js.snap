// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<div
  className="card card__large card__category wrapper"
  style={
    Object {
      "borderBottomColor": "#eee",
    }
  }
>
  <div
    className="iconWrapper"
  />
  <div
    className="text"
  />
</div>
`;
